pkg/operator/ceph/disruption/clusterdisruption/reconcile.go

```diff
@@ -257,3 +253,41 @@ func (c *ClusterMap) GetClusterNamespaces() []string {
 	}
 	return namespaces
 }
+
+func (r *ReconcileClusterDisruption) deleteDrainCanaryPods() error {
+	canaryLabels := client.MatchingLabels{k8sutil.AppAttr: "rook-ceph-drain-canary"}
+	drainCanaryList := &appsv1.DeploymentList{}
+	err := r.client.List(context.TODO(), drainCanaryList, canaryLabels, client.InNamespace(r.context.OperatorNamespace))
+	if err != nil && !kerrors.IsNotFound(err) {
+		return errors.Wrapf(err, "failed to list any drain canary pods by labels %q", canaryLabels)
+	}
+	if len(drainCanaryList.Items) > 0 {
+		logger.Info("deleting all legacy node drain canary pods")
+	}
+	for i := range drainCanaryList.Items {
+		err := r.client.Delete(context.TODO(), &drainCanaryList.Items[i])
+		if err != nil {
+			return errors.Wrapf(err, "failed to delete drain canary pod %q", drainCanaryList.Items[i].Name)
+		}
+	}
+	return nil
+}
+
+func (r *ReconcileClusterDisruption) deleteLegacyPDBForOSD() error {
+	legacyPDBLabels := client.MatchingLabels{k8sutil.AppAttr: "rook-ceph-osd-pdb"}
+	legacyPDBList := &policyv1beta1.PodDisruptionBudgetList{}
+	err := r.client.List(context.TODO(), legacyPDBList, legacyPDBLabels, client.InNamespace(r.context.OperatorNamespace))
```

Comment:
Instead of a `List()` then `Delete()`, can we call `Delete()` with the label? Then it can delete them all at the same time.

Reply:
using `DeleteAllOf`